src/wallet/wallet.cpp

```diff
@@ -2385,10 +2385,10 @@ CAmount CWallet::GetNormalizedAnonymizedBalance() const
 
     LOCK2(cs_main, cs_wallet);
     for (const auto& outpoint : setWalletUTXO) {
-        std::map<uint256, CWalletTx>::const_iterator it = mapWallet.find(outpoint.hash);
-        if (it == mapWallet.end()) continue;
-        if (!IsDenominated(outpoint)) continue;
-        if (it->second.GetDepthInMainChain() < 0) continue;
+        const auto it = mapWallet.find(outpoint.hash);
+
+        if (it == mapWallet.end() || it->second.GetDepthInMainChain() < 0) continue;
```

Comment:
`GetDepthInMainChain` is quite expensive compared to `IsDenominated/IsDenominatedAmount`. Maybe leave the order of checks as it was?